testutil/assert.go

```diff
@@ -0,0 +1,131 @@
+package testutil
+
+import (
+	"fmt"
+	"reflect"
+	"regexp"
+)
+
+type miniTB interface {
+	Helper()
+	Fatalf(string, ...any)
+}
+
+// formatSuffix builds an optional suffix from msgAndArgs
+func formatSuffix(msgAndArgs ...any) string {
+	switch len(msgAndArgs) {
+	case 0:
+		return ""
+	case 1:
+		return " - " + fmt.Sprintf("%+v", msgAndArgs[0])
+	default:
+		return " - " + fmt.Sprintf(msgAndArgs[0].(string), msgAndArgs[1:]...)
+	}
+}
+
+// Comparable types (numbers, strings, pointers to the same object, etc.).
+func AssertEqual[T comparable](tb miniTB, got, want T, msgAndArgs ...any) {
+	tb.Helper()
+	if got != want {
+		suffix := formatSuffix(msgAndArgs...)
+		tb.Fatalf("got %v; want %v%s", got, want, suffix)
+	}
+}
+
+// Anything else (slices, maps, structs with slices...).
+func AssertDeepEqual(tb miniTB, got, want any, msgAndArgs ...any) {
+	tb.Helper()
+	if !reflect.DeepEqual(got, want) {
+		suffix := formatSuffix(msgAndArgs...)
+		tb.Fatalf("got %+v; want %+v%s", got, want, suffix)
+	}
+}
+
+func AssertErrorMatches(tb miniTB, err error, pattern string, msgAndArgs ...any) {
```

Comment:
Here and everywhere, I would use two methods for each


```suggestion
func AssertErrorMatches(tb miniTB, err error, pattern string) {
```

And


```suggestion
func AssertErrorMatchesf(tb miniTB, err error, pattern string, msg string, args ...any) {
```

Avoiding the oddities with playing with type assertion

Reply:
You will find explanation here

https://github.com/Antonboom/testifylint?tab=readme-ov-file#3

Reply:
But I could live with what you implemented

Reply:
Not resolved

Reply:
I was working on it. It's done now, can you check?